pysmt/simplifier.py

```diff
@@ -280,47 +280,37 @@ def walk_exists(self, formula, args, **kwargs):
         return self.manager.Exists(varset, sf)
 
     def walk_plus(self, formula, args, **kwargs):
```

Comment:
This method was handling a few additional simplifications. 
Could you find or add a test that shows that old behaviors have been preserved?
Could you also add a test for showing that the handling of the algebraic numbers is correct?

Reply:
I have added a few very simple test cases, for both the simplification of Plus and the handling of algebraic numbers.   
The test cases for algebraic numbers highlighted a few other places in which they where not handled: TheoryOracle and MsatConverter.    
I tried to implement those, please have a look at them.    
* In TheoryOracle I simply consider them as real constants, also in the type checker the are always associated with REAL type. However, in principle a z3num.Numeral could also contain an integer value, I guess algebraic constants should never appear in pysmt-expressions of type INT, but I am not sure if, where and how this is enforced.
* In MsatConverter I handled z3num.Numeral that represent rational values. In the case of an irrational one a convert exception is raised. It could be possible to handle also this case by traversing the AST of the z3 numeral and building a corresponding msat expression. However, I haven't looked at how such ASTs are built and this might be part of a possibly larger "issue" about the representation of irrational numbers in pysmt.